src/data_info/services/sharing.clj

```diff
@@ -160,10 +161,15 @@
     (when-not (nil? matching-key)
       (string/replace-first path matching-key (mappings matching-key)))))
 
+(defn- encode-mapped-anon-path
+  "Take a path that's been appropriately mapped, and URL encode it for use. This function will use %20 for spaces."
+  [mapped-path]
+  (string/join "/" (map #(string/replace (URLEncoder/encode %) #"\+" "%20") (string/split mapped-path #"/"))))
+
 (defn anon-file-url
   [path]
   (let [aurl (url/url (cfg/anon-files-base-url))]
-    (str (-> aurl (assoc :path (apply ft/path-join (:path aurl) [(map-anon-url-path path)]))))))
+    (str (-> aurl (assoc :path (apply ft/path-join (:path aurl) [(encode-mapped-anon-path (map-anon-url-path path))]))))))
```

Comment:
Maybe it's just me, but the `->` threading only used for the `assoc` is making this hard to follow for me. I would go all out and thread everything, or not use threading at all here 😕 
```suggestion
    (-> aurl :path (ft/path-join (encode-mapped-anon-path (map-anon-url-path path))) (assoc :path))))
```

Reply:
I'm not sure if my suggestion is actually correct 🤔

Reply:
Hm, good point. I tried to change it as little as possible, as I generally do, but it's true it might be nice to thread it or otherwise break it up a bit more. It's certainly not written the most clearly it could be.

I guess the completely unthreaded version would be:

```clojure
(str
  (assoc aurl
         (apply ft/path-join
                (:path aurl)
                [(encode-mapped-anon-path
                   (map-anon-url-path path))])))
```
So I guess we could only thread `aurl` once more, into the `str` call, but maybe we could do better otherwise. I think using `apply` is extraneous, at least?

Maybe:

```clojure
(defn anon-file-url
  [path]
  (let [aurl (url/url (cfg/anon-files-base-url))
        encoded-path (encoded-mapped-anon-path (map-anon-url-path path))]
     (-> aurl
         (assoc :path (ft/path-join (:path aurl) encoded-path))
         str)))
```
... if I haven't typoed something there, anyway?

Reply:
Nice, LGTM! Even just moving the `encoded-path` into the `let` helps out 👍